lab_5_scrapper/scrapper.py

```diff
@@ -120,46 +253,184 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+        self.start_url = self._seed_urls[0]
+        self.all_urls = []
+        self._load_crawler_data()
+
+    def _load_crawler_data(self) -> None:
+        crawler_data_path = Path(__file__).parent / 'build' / 'crawler_data.json'
+
+        if crawler_data_path.exists():
+            with open(crawler_data_path, 'r', encoding='utf-8') as f:
+                crawler_data = json.load(f)
+            self.start_url = crawler_data['start_url']
+            self.urls = crawler_data['urls']
+            self.all_urls = crawler_data['all_urls']
+
+    def _save_crawler_data(self) -> None:
+        crawler_data = {
+            'start_url': self.start_url,
+            'urls': self.urls,
+            'all_urls': self.all_urls
+        }
+        crawler_data_path = Path(__file__).parent / 'build' / 'crawler_data.json'
+        with open(crawler_data_path, 'w', encoding='utf-8') as f:
+            json.dump(crawler_data, f, ensure_ascii=True, indent=4, separators=(', ', ': '))
+
+    def find_articles(self) -> None:
+        """
+        Finds articles
+        """
+        res = make_request(self.start_url, self._config)
+        soup = BeautifulSoup(res.content, 'lxml')
+
+        relevant_urls = [*map(self._extract_url, soup.find_all('a', class_=(
+            'tape-list-item', 'small-mix-item', 'small-line-item')))]
+
+        for url in relevant_urls:
+            if len(self.urls) >= self._config.get_num_articles():
+                return
+            if not url or url in self.all_urls:
+                continue
+            self.all_urls.append(url)
+            if url in relevant_urls:
+                self.urls.append(url)
+                print(url)
+            self.start_url = url
+            self._save_crawler_data()
+            self.find_articles()
+
+
+conf = Config(path_to_config=CRAWLER_CONFIG_PATH)
 
 
 def main() -> None:
     """
     Entrypoint for scrapper module
     """
-    pass
+    prepare_environment(ASSETS_PATH)
+    crawler = Crawler(config=conf)
+    crawler.find_articles()
+    for ind, url in enumerate(crawler.urls, start=1):
+        parser = HTMLParser(full_url=url, article_id=ind, config=conf)
+        article = parser.parse()
+        if isinstance(article, Article):
+            to_raw(article)
+            to_meta(article)
+
+
+def main_recursive() -> None:
+    """
+    Driver code for recursive crawling
+    """
+    prepare_environment(ASSETS_PATH)
+    crawler_recursive = CrawlerRecursive(config=conf)
+    crawler_recursive.find_articles()
+    for i, url in enumerate(crawler_recursive.urls, start=1):
+        crawler_recursive.find_articles()
+        parser = HTMLParser(full_url=url, article_id=i, config=conf)
+        article = parser.parse()
+        if isinstance(article, Article):
+            to_raw(article)
+            to_meta(article)
 
 
 if __name__ == "__main__":
-    main()
+    if conf.get_headless_mode():
+        main_recursive()
+    else:
+        main()
```

Comment:
just call main function in this block

what is the meaning to use condition and headless_mode to switch implementations?